src/components/headersEditor/UrlPreview.tsx

```diff
@@ -0,0 +1,39 @@
+import { CopyOutlined } from '@ant-design/icons';
+import { Button, Col, Divider, Form, Input } from 'antd';
+import { Row } from 'antd/lib';
+import { useState, useEffect } from 'react';
+
+import { UrlPreviewProps } from '@/types/headersSectionTypes';
+
+export default function UrlPreview({ endpoint, headers }: Readonly<UrlPreviewProps>) {
+  const [fullUrl, setFullUrl] = useState<string>(endpoint);
+
+  useEffect(() => {
+    const headerParams = Object.entries(headers)
+      .map(([key, value]) => `${encodeURIComponent(key)}=${encodeURIComponent(value)}`)
+      .join('&');
+
+    const urlWithHeaders = headerParams ? `${endpoint}?${headerParams}` : endpoint;
+    setFullUrl(urlWithHeaders);
+  }, [endpoint, headers]);
```

Comment:
можно упростить, нам же вся эта конструкция нужна только для того, чтоб получить `fullUrl`

```suggestion
  const fullUrl = useMemo(() => {
    const headerParams = Object.entries(headers)
      .map(([key, value]) => `${encodeURIComponent(key)}=${encodeURIComponent(value)}`)
      .join('&');
    return headerParams ? `${endpoint}?${headerParams}` : endpoint;
  }, [endpoint, headers]);
```

Reply:
исправила, спасибо)